src/oneTrust/helpers/syncOneTrustAssessments.ts

```diff
@@ -0,0 +1,113 @@
+import { Got } from 'got/dist/source';
+import {
+  getListOfOneTrustAssessments,
+  getOneTrustAssessment,
+  getOneTrustRisk,
+} from '../endpoints';
+import { map, mapSeries } from 'bluebird';
+import { logger } from '../../logger';
+import {
+  OneTrustAssessmentQuestion,
+  OneTrustAssessmentSection,
+  OneTrustGetRiskResponse,
+} from '@transcend-io/privacy-types';
+import uniq from 'lodash/uniq';
+import { enrichOneTrustAssessment } from './enrichOneTrustAssessment';
+import { syncOneTrustAssessmentToDisk } from './syncOneTrustAssessmentToDisk';
+import { OneTrustFileFormat } from '../../enums';
+import { GraphQLClient } from 'graphql-request';
+import { syncOneTrustAssessmentToTranscend } from './syncOneTrustAssessmentToTranscend';
+
+export interface AssessmentForm {
+  /** ID of Assessment Form */
+  id: string;
+  /** Title of Assessment Form */
+  name: string;
+}
+
+export const syncOneTrustAssessments = async ({
+  oneTrust,
+  file,
+  fileFormat,
+  dryRun,
+  transcend,
+}: {
+  /** the OneTrust client instance */
+  oneTrust: Got;
+  /** the Transcend client instance */
+  transcend?: GraphQLClient;
+  /** Whether to write to file instead of syncing to Transcend */
+  dryRun: boolean;
+  /** the path to the file in case dryRun is true */
+  file?: string;
+  /** the format of the file in case dryRun is true */
+  fileFormat?: OneTrustFileFormat;
+}): Promise<void> => {
+  // fetch the list of all assessments in the OneTrust organization
+  logger.info('Getting list of all assessments from OneTrust...');
+  const assessments = await getListOfOneTrustAssessments({ oneTrust });
+
+  /**
+   * fetch details about each assessment in series and write to transcend or to disk
+   * (depending on the dryRun argument) right away to avoid running out of memory
+   */
+  await mapSeries(assessments, async (assessment, index) => {
```

Comment:
wish we could do this with some concurrency.. maybe chunks of 5-10 assessments at a time? IDK what their rate limit is but we'd want to stay under it

Reply:
I think OT would allow for it. My main concern was introducing concurrency bugs (e.g., writing to a file is pretty slow and it's possible to mess up the writing order if doing so concurrently).

Reply:
Perhaps we can do the fetching from OneTrust concurrently but the writing to Disk or Transcend synchronously? I. can try that.

Reply:
^ yea! that's sort of what i was thinking. Concurrently fire off 5-10 requests but then write it down concurrently